BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,341 @@
+            aGameRecord[kSpatialSpanMemoryGameRecordSeedKey]      = NSNumber(unsignedInt: aRecord.seed)
+            aGameRecord[kSpatialSpanMemoryGameRecordGameSizeKey]  = aRecord.gameSize
+            aGameRecord[kSpatialSpanMemoryGameRecordGameScoreKey] = aRecord.score
+            aGameRecord[kSpatialSpanMemoryGameRecordSequenceKey]  = aRecord.sequence
+            aGameRecord[kSpatialSpanMemoryGameStatusKey]          = gameStatusKeys[aRecord.gameStatus.rawValue]
+            
+            let touchSamples = makeTouchSampleRecords(aRecord.touchSamples)
+            aGameRecord[kSpatialSpanMemoryGameRecordTouchSamplesKey] = touchSamples
+            
+            let rectangles = makeTargetRectangleRecords(aRecord.targetRects!)
+            aGameRecord[kSpatialSpanMemoryGameRecordTargetRectsKey] = rectangles
+            
+            gameRecords += [aGameRecord]
+        }
+        memoryGameResults[kSpatialSpanMemorySummaryGameRecordsKey] = gameRecords
+        let filename = bridgifyFilename(self.identifier) + ".json"
+        memoryGameResults[kItemKey] = filename
+        return ArchiveableResult.init(result: (memoryGameResults as NSDictionary).jsonObject(), resultType: .Dictionary, filename: filename)
+    }
+    
+    
+    func makeTouchSampleRecords(touchSamples: [ORKSpatialSpanMemoryGameTouchSample]?) -> NSArray
+    {
+        var samples = [[String: AnyObject]]()
+        
+        guard touchSamples != nil else {
+            return samples
+        }
+        for sample: ORKSpatialSpanMemoryGameTouchSample in touchSamples! {
+            
+            var aTouchSample = [String: AnyObject]()
+            
+            aTouchSample[kSpatialSpanMemoryTouchSampleTimeStampKey]   = sample.timestamp
+            aTouchSample[kSpatialSpanMemoryTouchSampleTargetIndexKey] = sample.targetIndex
+            aTouchSample[kSpatialSpanMemoryTouchSampleLocationKey]    = NSStringFromCGPoint(sample.location)
+            aTouchSample[kSpatialSpanMemoryTouchSampleIsCorrectKey]   = sample.correct
+            
+            samples += [aTouchSample]
+        }
+        return  samples
+    }
+    
+    func makeTargetRectangleRecords(targetRectangles: [NSValue]) -> NSArray
+    {
+        var rectangles = [String]()
+        
+        for value: NSValue in targetRectangles {
+            let rectangle = value.CGRectValue()
+            let stringified = NSStringFromCGRect(rectangle)
+            rectangles += [stringified]
+        }
+        return  rectangles
+    }
+}
+
+class AnswerKeyAndValue: NSObject {
+    var key: String
+    var value: AnyObject?
+    
+    init(key: String, value: AnyObject?) {
+        self.key = key
+        self.value = value
+    }
+}
+
+protocol ORKQuestionResultAnswerJSON {
+    func jsonSerializedAnswer() -> (AnswerKeyAndValue)?
+}
+
+private var _timeOnlyFormatter: NSDateFormatter? = nil
+
+extension ORKQuestionResult: ORKQuestionResultAnswerJSON {
+    
+    func timeOnlyFormatter() -> NSDateFormatter {
```

Comment:
It doesn't yet, actually, but it probably makes sense to move this there anyway. I put it here partly because BridgeSDK doesn't actually use this internally, but mostly because I was in a hurry.